financial-connections/src/main/java/com/stripe/android/financialconnections/features/linkaccountpicker/LinkAccountPickerViewModel.kt

```diff
@@ -290,23 +292,26 @@ internal class LinkAccountPickerViewModel @AssistedInject constructor(
 
         val accounts = requireNotNull(stateFlow.value.payload()?.accounts)
         val drawerOnSelection = accounts.firstOrNull { it.account.id == partnerAccount.id }?.display?.drawerOnSelection
-
-        if (drawerOnSelection != null) {
-            presentSheet(
-                content = NoticeSheetState.NoticeSheetContent.Generic(drawerOnSelection),
-                referrer = PANE,
-            )
-            return
-        }
-
-        val updateRequired = createUpdateRequiredContent(
+        val updateRequired: AccountUpdateRequiredState.Payload? = createUpdateRequiredContent(
             partnerAccount = partnerAccount,
+            // Use selected account icon (not coming on the SDU response as selection is not known by backend)
+            genericContent = drawerOnSelection?.withIcon(partnerAccount.institution?.icon?.default),
             partnerToCoreAuths = stateFlow.value.payload()?.partnerToCoreAuths,
         )
 
         if (updateRequired != null) {
             logUpdateRequired(updateRequired)
-            presentUpdateRequiredSheet(updateRequired, referrer = PANE)
+            presentUpdateRequiredSheet(
+                updateRequired,
+                referrer = PANE,
+            )
+            return
+        }
+        if (drawerOnSelection != null) {
+            presentSheet(
+                content = Generic(drawerOnSelection),
+                referrer = PANE,
+            )
```

Comment:
Changed order here as other platforms do - since `drawerOnSelection` is also used for the update required content, we need to check for update required conditions before falling back to `drawerOnSelection` existence.